src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -34,7 +35,7 @@ internal bool NavAnimationInProgress
 					return;
 				_navAnimationInProgress = value;
 				if (value)
-					MessagingCenter.Send(this, CloseContextActionsSignalName);
+					WeakReferenceMessenger.Default.Send(new CloseContextActionsMessage());
```

Comment:
Nit: could be simplified:

```suggestion
					WeakReferenceMessenger.Default.Send<CloseContextActionsMessage>();
```

Reply:
Cool, I didn't know that it could automatically create the message for me. 
Question, though - are there performance implications here? Is this using Activator.CreateInstance to create the message object?